tests/drivers/fuel_gauge/lc709203f/src/test_lc709203f.c

```diff
@@ -0,0 +1,195 @@
+
+	zassert_equal(ret, -ENOTSUP);
+}
+
+ZTEST_USER_F(lc709203f, test_set_some_props_failed__returns_err)
+{
+	fuel_gauge_prop_t prop_types[] = {
+		/* First invalid property */
+		FUEL_GAUGE_PROP_MAX,
+		/* Second invalid property */
+		FUEL_GAUGE_PROP_MAX,
+		/* Valid property */
+		FUEL_GAUGE_STATE_OF_CHARGE_ALARM,
+		/* Set Manufacturer's Access to arbitrary word */
+
+	};
+
+	union fuel_gauge_prop_val props[] = {
+		/* First invalid property */
+		{0},
+		/* Second invalid property */
+		{0},
+		/* Valid property */
+		/* Sets state of charge threshold to generate Alarm signal*/
+		{.state_of_charge_alarm = 10},
+	};
+
+	int ret = fuel_gauge_set_props(fixture->dev, prop_types, props, ARRAY_SIZE(props));
+
+	zassert_equal(ret, -ENOTSUP);
+}
+
+ZTEST_USER_F(lc709203f, test_set_prop_can_be_get)
+{
+	uint16_t sbs_mode = 0x0002;
+	uint16_t current_direction = 0x0001;
+	uint8_t state_of_charge_alarm = 20;
+	uint32_t low_voltage_alarm = 3000 * 1000;
+
+	fuel_gauge_prop_t prop_types[] = {
+		FUEL_GAUGE_SBS_MODE,
+		FUEL_GAUGE_CURRENT_DIRECTION,
+		FUEL_GAUGE_STATE_OF_CHARGE_ALARM,
+		FUEL_GAUGE_LOW_VOLTAGE_ALARM,
+	};
+
+	union fuel_gauge_prop_val set_props[] = {
+		{
+			.sbs_mode = sbs_mode,
+		},
+		{
+			.current_direction = current_direction,
+		},
+		{
+			.state_of_charge_alarm = state_of_charge_alarm,
+		},
+		{
+			.low_voltage_alarm = low_voltage_alarm,
+		},
+	};
+
+	union fuel_gauge_prop_val get_props[ARRAY_SIZE(prop_types)];
+
+	zassert_ok(
+		fuel_gauge_set_props(fixture->dev, prop_types, set_props, ARRAY_SIZE(set_props)));
+
+	zassert_ok(
+		fuel_gauge_get_props(fixture->dev, prop_types, get_props, ARRAY_SIZE(get_props)));
+
+	zassert_equal(get_props[0].sbs_mode, sbs_mode);
+	zassert_equal(get_props[1].current_direction, current_direction);
+	zassert_equal(get_props[2].state_of_charge_alarm, state_of_charge_alarm);
+	zassert_equal(get_props[3].low_voltage_alarm, low_voltage_alarm);
+}
+
+ZTEST_USER_F(lc709203f, test_get_props__returns_ok)
+{
+	fuel_gauge_prop_t props[] = {
+		FUEL_GAUGE_RELATIVE_STATE_OF_CHARGE,
+		FUEL_GAUGE_TEMPERATURE,
+		FUEL_GAUGE_VOLTAGE,
+		FUEL_GAUGE_SBS_MODE,
+		FUEL_GAUGE_DESIGN_CAPACITY,
+		FUEL_GAUGE_CURRENT_DIRECTION,
+		FUEL_GAUGE_STATE_OF_CHARGE_ALARM,
+		FUEL_GAUGE_LOW_VOLTAGE_ALARM,
+	};
+	union fuel_gauge_prop_val vals[ARRAY_SIZE(props)];
+
+	int ret = fuel_gauge_get_props(fixture->dev, props, vals, ARRAY_SIZE(props));
+
+#if CONFIG_EMUL
+	zassert_equal(vals[0].relative_state_of_charge, 50);
+	zassert_equal(vals[1].temperature, 0x0BA6);
+	zassert_equal(vals[2].voltage, 3700 * 1000);
+	zassert_equal(vals[3].sbs_mode, 0x0001);
+	zassert_equal(vals[4].design_cap, 500);
+	zassert_true(((vals[5].current_direction == 0x0000) ||
+		      (vals[5].current_direction == 0x0001) ||
+		      (vals[5].current_direction == 0xFFFF)));
+	zassert_equal(vals[6].state_of_charge_alarm, 0x0008);
+	zassert_equal(vals[7].state_of_charge_alarm, 0x0000);
```

Comment:
typo -- but prob means there's a bug somewhere if it wasn't caught when running the tests?

```suggestion
	zassert_equal(vals[7]. low_voltage_alarm, 0x0000);
```